File: ios/Classes/AzureCall.swift
import AzureCommunicationCalling
import AzureCommunicationUICalling

class AzureCall {
    func startCallComposite(token: String, roomId: String, displayName: String?) -> String? {
        do {
            let callCompositeOptions = CallCompositeOptions(displayName: displayName)
            let communicationTokenCredential = try CommunicationTokenCredential(token: token)
            
            let callComposite = CallComposite(credential: communicationTokenCredential, withOptions: callCompositeOptions)
            
            let locator = JoinLocator.roomCall(roomId: roomId)
            
            callComposite.launch(locator: locator)
        } catch {
            return "\(error)";
        }
        
        return nil
    }
}

File: android/src/main/kotlin/com/arashz4/azure_communication_calling/AzureCall.kt
package com.arashz4.azure_communication_calling

import android.content.Context
import com.azure.android.communication.common.CommunicationTokenCredential
import com.azure.android.communication.ui.calling.CallComposite
import com.azure.android.communication.ui.calling.CallCompositeBuilder
import com.azure.android.communication.ui.calling.models.CallCompositeRoomLocator

class AzureCall {
    fun startCallComposite(context: Context, token: String, roomId: String, displayName: String?) : String? {
        try {
            val credential = CommunicationTokenCredential(token);

            val callComposite: CallComposite = CallCompositeBuilder()
                .applicationContext(context)
                .credential(credential)
                .displayName(displayName).build()

            val locator = CallCompositeRoomLocator(roomId)

            callComposite.launch(context, locator)
        } catch (e: Exception) {
            return e.message;
        }

        return null;
    }
}
